govcd/vapptemplate.go

```diff
@@ -65,12 +65,42 @@ func (vAppTemplate *VAppTemplate) Refresh() error {
 	vAppTemplate.VAppTemplate = &types.VAppTemplate{}
 
 	_, err := vAppTemplate.client.ExecuteRequest(url, http.MethodGet,
-		"", "error retrieving vApp template item: %s", nil, vAppTemplate.VAppTemplate)
+		"", "error retrieving vApp Template: %s", nil, vAppTemplate.VAppTemplate)
 
 	return err
 }
 
-// Update updates the vApp template item information
+// GetCatalogName gets the catalog name to which the receiver vApp Template belongs
+func (vAppTemplate *VAppTemplate) GetCatalogName() (string, error) {
+	queriedVappTemplates, err := queryVappTemplateListWithFilter(vAppTemplate.client, map[string]string{
+		"id": vAppTemplate.VAppTemplate.ID,
+	})
+	if err != nil {
+		return "", err
+	}
+	if len(queriedVappTemplates) != 1 {
+		return "", fmt.Errorf("found more than one vApp Template with ID %s", vAppTemplate.VAppTemplate.ID)
+	}
+	return queriedVappTemplates[0].CatalogName, nil
+}
+
+// GetVdcName gets the VDC id to which the receiver vApp Template belongs
+func (vAppTemplate *VAppTemplate) GetVdcName() (string, error) {
+	queriedVappTemplates, err := queryVappTemplateListWithFilter(vAppTemplate.client, map[string]string{
+		"id": vAppTemplate.VAppTemplate.ID,
+	})
+	if err != nil {
+		return "", err
+	}
+	if len(queriedVappTemplates) != 1 {
+		return "", fmt.Errorf("found more than one vApp Template with ID %s", vAppTemplate.VAppTemplate.ID)
+	}
+	return queriedVappTemplates[0].VdcName, nil
+}
+
+// Update updates the vApp template item information.
+// VCD also updates the associated Catalog Item, in order to be in sync with the receiver vApp Template entity.
+// For example, updating a vApp Template name "A" to "B" will make VCD to also update the Catalog Item to be renamed to "B".
```

Comment:
Does this mean if someone has both item and template in the same HCL, and changes template details, the next plan will show inconsistency for the item?

CC: @dataclouder @Didainius

Reply:
having "both item and template in the same HCL" would be pointless. If someone does, there will be an inconsistency, but this is why we have a cleaner game, compared to the UI, where information can be hidden and convergence of entities presented as a unified view.
When we will eventually deprecate catalog_item, this problem will disappear. For now, users know that they need either of them. If they want to break the provider, there are other similar places that offer such opportunity.
IMO, this is a non-problem.

Reply:
I concur. It would actually be good to have a runtime sanity check that the user is not using both types for the same entity, but given Terraform limitations that would be difficult.

Reply:
Ideally it should be very hard to achieve as both `vcd_catalog_item` and `vcd_catalog_vapp_template` require a name and catalog, which must be a unique pair in VCD. So, if someone uses both in same HCL, one of them should give an error on creation saying that the vApp Template exists already.

To bypass that one could use the importing feature, I guess? In that case an inconsistency could be achieved.